src/test/unit/io/json/json_data_tuples_test.cpp

```diff
@@ -0,0 +1,180 @@
+#include <stan/io/json/json_data.hpp>
+#include <stan/io/json/json_data_handler.hpp>
+#include <stan/io/json/json_error.hpp>
+#include <stan/io/json/json_handler.hpp>
+#include <stan/io/json/rapidjson_parser.hpp>
+
+#include <test/unit/util.hpp>
+#include <test/unit/io/json/util.hpp>
+
+#include <boost/limits.hpp>
+#include <boost/math/concepts/real_concept.hpp>
+#include <boost/math/special_functions/fpclassify.hpp>
+#include <gtest/gtest.h>
+#include <complex>
+#include <iostream>
+#include <fstream>
+
+// tuple(int, real) x - also real y;
```

Comment:
Do we have a way of testing bad inputs?

I noticed some worse-than-before errors for things like repeated keys, e.g. if you changed this test file to

```json
{ "x": {"1": 1, "2": 6.28},
"y": 0.0000123,
"x": 0.0000123
}
```

The test still passes (e.g. redefining `x` does not error out).

Similarly, re-defining an name which previously defined an array-of-tuples yields ` error: non-rectangular array.` which isn't really the right error for this. To see this, update `array_tuple_multi.json` to also have an `"x": []` element.

Reply:
problem is this:  the tuple results in Stan program vars "x.1" and "x.2"
the scalar var is just "x"
code needs to check that prefix is unique.   will add logic for this.

Reply:
It should be valid to also insert “x” as an (empty) variable into the map, if that would get us the errors “for free”